src/main/java/org/apache/maven/plugins/pmd/exec/PmdErrorMessageReporter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.plugins.pmd.exec;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.sourceforge.pmd.util.log.PmdReporter;
+import org.apache.maven.reporting.MavenReportException;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.slf4j.event.Level;
+
+class PmdErrorMessageReporter implements PmdReporter {
+    private final PmdReporter delegate;
+    private List<Throwable> errors = new ArrayList<>();
+
+    PmdErrorMessageReporter(PmdReporter delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public boolean isLoggable(Level level) {
+        return delegate.isLoggable(level);
+    }
+
+    @Override
+    public void logEx(Level level, @Nullable String message, Object[] formatArgs, @Nullable Throwable error) {
```

Comment:
logEx --> logException

Reply:
That can't be changed here, as that's an interface method to be implemented.